src/middleware/validation.ts

```diff
@@ -0,0 +1,256 @@
+import bundle from './validation.nls';
+import { create } from '@dojo/framework/core/vdom';
+import i18n from '@dojo/framework/core/middleware/i18n';
+
+const upperCaseRegEx = /[A-Z]/;
+const numbersRegEx = /\d/;
+
+export interface ValidationRules {
+	length?: {
+		min?: number;
+		max?: number;
+	};
+	contains?: {
```

Comment:
If you specify `contains`, I think at least one of the options should be specified?

Reply:
was thinking something like this: http://www.typescriptlang.org/play/#code/JYOwLgpgTgZghgYwgAgDIRAczACwLKgBKArgDYoDeAUMrcgLagBcyIx9ARtANw131wAHgH4WbTjyoBfKlVCRYiFOiy48QkuWTU6DUKNbsuUXroGCxRyTLnho8JMgDCAe3BxQAZwCCYdHE8wbT5aOD8IALBLCRMQ5GIABwToBACIA3FjUzpM6E8Mq1jdT2SEYDhSJxw4KEQFfOis6Vl5eyVnNzAPEE8AVSToJzTg3UTkqFTPCEbJXTD-QIKY7NpcqAbDZbiSiDKKqpq6vKWmm1bFR1d3LwA5Qs8RnPuZorp5iMWXlfiBibST2Z0HZ7SrVWoIeoA2JnOwXFBXLpeADKpXKoMOELyj1owLRB3B9S+cXekSh3zGKX+RN0aw2a14NggggSLigQXODhQADUKsAACZhYBuTQQB7UACQ5FUOAMKmw+CIZBQAB80Bh5epBCLeOKEJ1uhsEQbfAsgqqjbd7shzfrkaj9mCjutrR1rj1+uMhlMGbI9T0gnAWDzSPzBcKlQ8ALzY5BS+UsCgyGx+wLIDhB3kCsBCkAiqMxuO4BNxMzMZAARjiSd9blTCAzIazObzyGjOjohZwxd0ZiELErumrVBTQT5DdD2fD5Hz7done7Pdo5n7AAYADQl-hlgd0Icj5DTZDBifNiOtmN+xE9BN72tBGDjptT0Xn2fIS8GheLkmBftV5rDneyCYI+Ya5mebZxB+Xhfj2tJ-oOAH7l2R6ZmBLaQbo0HXjGxT2uiBJ5Ahu5IUBwCgZO4HTq+UG2jhb6jL8kyHjutC3v6yDAAAVhRp7UZhdDYZ4sGMeMzH9hui6rM8yAAMz-lIQA

Thanks ts playground for the massive url

Reply:
Done! Much nicer types ❤️